### PR TITLE
[ENH] Add FreTS v2 model

### LINKED ISSUE #2191: [ENH] Add FreTS v2 implementation (Frequency-domain MLPs for Time Series)
**Is your feature request related to a problem? Please describe.**
Add more v2 models to pytorch-forecasting monorepo. And pytorch-forecasting v2 has no frequency-domain model yet.

**Describe the solution you'd like**
Add FreTS as a v2 model. It's a small and simple model. 

**Describe alternatives you've considered**
DLinear/NLinear are already in the library but purely linear. FreTS is more expressive while still being lightweight.

**Additional context**
Reference: https://github.com/thuml/Time-Series-Library/blob/main/models/FreTS.py

### PR DESCRIPTION
**Pull request by echo-xiao**
#### Reference Issues/PRs
Close https://github.com/sktime/pytorch-forecasting/issues/2191

#### What does this implement/fix? Explain your changes.
Adds a v2 implementation of the FreTS model for long-term time series forecasting.

#### What should a reviewer concentrate their feedback on?

  - Correctness of the FreMLP implementation in FreTSCore
  - Whether channel_independence=False is correctly handled
  - The choice of using target_past as model input and metadata["target"] for n_channels

#### Did you add any tests for the change?

  Yes. tests/test_models/test_frets_v2.py

#### PR checklist

- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.
- [x] Added/modified tests
- [x] Used pre-commit hooks when committing to ensure that code is compliant with hooks. Install hooks with `pre-commit install`.

### CODE REVIEW

File: pytorch_forecasting/models/frets/_frets_v2.py [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.optim import Optimizer
+
+from pytorch_forecasting.metrics import MAE, Metric
+from pytorch_forecasting.models.base._base_model_v2 import BaseModel
+
+
+class FreTSCore(nn.Module):
```
Comment (phoeenniixx) [inline pytorch_forecasting/models/frets/_frets_v2.py]: Should this go to `layers` module?

File: pytorch_forecasting/models/frets/_frets_pkg_v2.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
+"""FreTS v2 package container."""
+
+from pytorch_forecasting.base._base_pkg import Base_pkg
+
+
+class FreTS_v2_pkg_v2(Base_pkg):
```
Comment (phoeenniixx) [inline pytorch_forecasting/models/frets/_frets_pkg_v2.py]: I think it should be `FreTS_pkg_v2`?

**Review by phoeenniixx — CHANGES_REQUESTED** [review]
Thanks!
For tests, you can also directly use `pkg` class for `fit` ,`predict` and no need to create separate instances of the datamodule and models, I think `pkg` can do that automatically, you just need to pass the cfgs. See `_integration` test [here](https://github.com/sktime/pytorch-forecasting/blob/main/pytorch_forecasting/tests/test_all_v2/_test_integration.py) to see what I mean.

This part is not documented clearly, so if you feel any issue lmk! We are working on improving the docs

**Comment by echo-xiao** [comment]
hi @phoeenniixx , update on the PR, please take a look.

1. reworked the tests.
2. moved FreTSCore into layers/
3. cleaned up the naming

File: tests/test_models/test_frets_v2.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    model_cfg.setdefault("loss", SMAPE())
+    return FreTS_pkg_v2(
+        model_cfg=model_cfg,
+        trainer_cfg=TRAINER_CFG,
+        datamodule_cfg=dm_cfg,
+    )
+
+
+@pytest.mark.parametrize("model_cfg", FreTS_pkg_v2.get_test_train_params())
+def test_frets_v2_integration(test_data, model_cfg):
```
Comment (phoeenniixx) [inline tests/test_models/test_frets_v2.py]: the test framework already takes care of this test

File: tests/test_models/test_frets_v2.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    assert isinstance(predictions, dict)
+    assert "prediction" in predictions
+
+    pred = predictions["prediction"]
+    assert isinstance(pred, torch.Tensor)
+    assert pred.ndim == 3, f"prediction must be 3D, got {pred.ndim}D"
+    assert pred.shape[1] == expected_pred_len
+
+
+def test_frets_v2_pkg_get_cls():
```
Comment (phoeenniixx) [inline tests/test_models/test_frets_v2.py]: same for this

**Review by phoeenniixx — CHANGES_REQUESTED** [review]
Thanks!
I think we dont need the `test_frets_v2` file? those tests are being handled by the test framework already

Also, please add the API reference here: https://github.com/sktime/pytorch-forecasting/blob/main/docs/source/m_layer_v2.rst#api-reference

File: tests/test_models/test_frets_v2.py (hunk 4 above)
Comment (echo-xiao) [reply tests/test_models/test_frets_v2.py]: Hi @phoeenniixx, thanks for the review!

- Removed tests/test_models/test_frets_v2.py file
- Added the API reference for `FreTS`

**Comment by agobbifbk** [comment]
Thx for adding this, just one comment: I see you use only the target past, it worth nothing to also use all the past covariates and add a final layer that maps the dimension to the target dimension, what do you think? Otherwise this will be a little bit week as model no?

**Comment by echo-xiao** [comment]
thx @agobbifbk , you are right.

- concat endor_cont with target_past + final linear back to target dim
- exogenous=True

**Review by phoeenniixx — CHANGES_REQUESTED** [review]
Please resolve the merge conflicts

**Comment by echo-xiao** [comment]
@phoeenniixx merge conflicts solved. thank you.

**Comment by Muhammad-Rebaal** [comment]
Hi @echo-xiao,

I think FreTSCore should be in `layers/_block` as **_frets_block** not in a new dir `_frets` and also add it up here : https://github.com/sktime/pytorch-forecasting/blob/main/docs/source/pkg_v2.rst

**Comment by echo-xiao** [comment]
> Hi @echo-xiao,
> 
> I think FreTSCore should be in `layers/_block` as **_frets_block** not in a new dir `_frets` and also add it up here : https://github.com/sktime/pytorch-forecasting/blob/main/docs/source/pkg_v2.rst

Hi, @Muhammad-Rebaal yes, you are right. Thanks for your review. I have already updated the code.

**Review by phoeenniixx — APPROVED** [review]
Thanks
